include/host/wasi/vinode.h

```diff
@@ -676,7 +696,8 @@ class VPoller : private Poller {
 
   WasiExpect<void> read(std::shared_ptr<VINode> Fd,
                         __wasi_userdata_t UserData) noexcept {
-    if (!Fd->can(__WASI_RIGHTS_POLL_FD_READWRITE | __WASI_RIGHTS_FD_READ)) {
+    if (!Fd->can(__WASI_RIGHTS_POLL_FD_READWRITE) &&
```

Comment:
Why does the `read` function need `POLL_FD_READWRITE` capability?
Should it be `OR` relationship.

Reply:
yes, `read` need `POLL_FD_READWRITE `  `OR`  `__WASI_RIGHTS_FD_READ`.

```
WasiExpect<void> read(std::shared_ptr<VINode> Fd,
                        __wasi_userdata_t UserData) noexcept {
    if (!Fd->can(__WASI_RIGHTS_POLL_FD_READWRITE) &&
        !Fd->can(__WASI_RIGHTS_FD_READ)) {
      return WasiUnexpect(__WASI_ERRNO_NOTCAPABLE);
    }
    return Poller::read(Fd->Node, UserData);
}
```
same as
```
WasiExpect<void> read(std::shared_ptr<VINode> Fd,
                        __wasi_userdata_t UserData) noexcept {
    if (Fd->can(__WASI_RIGHTS_POLL_FD_READWRITE | __WASI_RIGHTS_FD_READ){
        return Poller::read(Fd->Node, UserData);
    }
    return WasiUnexpect(__WASI_ERRNO_NOTCAPABLE);
}
```